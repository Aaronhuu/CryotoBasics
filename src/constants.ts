// API Configuration
export const API_CONFIG = {
  BASE_URL: "http://localhost:3001/api",
  PORT: 3001,
} as const;

// Server Status
export const SERVER_STATUS = {
  CHECKING: "checking",
  ONLINE: "online",
  OFFLINE: "offline",
} as const;

// UI Text
export const UI_TEXT = {
  APP_TITLE: "Crypto Basics Demo",
  DATABASE_RECORDS: (count: number) => `Database: ${count} records`,
  SERVER_STATUS: (status: string) => `Server: ${status}`,
  DATABASE_STATUS: (count: number, status: string) =>
    `Database: ${count} records | Server: ${status}`,
  LOADING: "Loading...",
  BACK: "← Back",
} as const;

// Button Labels
export const BUTTON_LABELS = {
  CLEAR_DATABASE: "Clear Database",
  INSERT_DATA: "Insert Data",
  SHOW_CRYPTO_LIST: "Show Crypto List",
  SHOW_FIAT_LIST: "Show Fiat List",
  SHOW_PURCHASABLE: "Show Purchasable",
} as const;

// Modal Titles
export const MODAL_TITLES = {
  CRYPTO_LIST: "Currency List A - Crypto",
  FIAT_LIST: "Currency List B - Fiat",
  PURCHASABLE_LIST: "Purchasable Currencies (A & B)",
} as const;

// Alert Messages
export const ALERT_MESSAGES = {
  DATABASE_CLEARED: {
    title: "Database Cleared",
    message: "All data has been removed from the local database.",
  },
  DATA_INSERTED: {
    title: "Data Inserted",
    message: (count: number) =>
      `${count} currency records have been inserted into the local database.`,
  },
  SERVER_OFFLINE: {
    title: "Server Offline",
    message: "Please start the server first using 'npm run server'",
  },
  NO_DATA: {
    title: "No Data",
    message: "Please insert data first using the 'Insert Data' button.",
  },
  ERRORS: {
    CLEAR_DATABASE: {
      title: "Error",
      message: "Failed to clear database.",
    },
    INSERT_DATA: {
      title: "Error",
      message: "Failed to insert data. Make sure the server is running.",
    },
    LOAD_CRYPTO: {
      title: "Error",
      message: "Failed to load crypto currencies.",
    },
    LOAD_FIAT: {
      title: "Error",
      message: "Failed to load fiat currencies.",
    },
    LOAD_PURCHASABLE: {
      title: "Error",
      message: "Failed to load purchasable currencies.",
    },
  },
} as const;

// Currency Types
export const CURRENCY_TYPES = {
  CRYPTO: "crypto",
  FIAT: "fiat",
} as const;

// API Messages
export const API_MESSAGES = {
  CRYPTO_SUCCESS: "Crypto currencies retrieved successfully",
  FIAT_SUCCESS: "Fiat currencies retrieved successfully",
  ALL_SUCCESS: "All currencies retrieved successfully",
  SEARCH_RESULTS: (term: string) => `Search results for "${term}"`,
  CURRENCY_FOUND: (id: string) => `Currency ${id} found`,
  CURRENCY_NOT_FOUND: (id: string) => `Currency ${id} not found`,
  SERVER_RUNNING: "Server is running",
  FILE_READ_ERROR: "File read error",
  LOAD_FAILED: {
    CRYPTO: "Failed to load crypto currencies",
    FIAT: "Failed to load fiat currencies",
    ALL: "Failed to load currencies",
    SEARCH: "Failed to search currencies",
  },
} as const;

// Search Placeholders
export const SEARCH_TEXT = {
  PLACEHOLDER: "Search currencies...",
  NO_RESULTS: "No currencies found",
  TRY_DIFFERENT: "Try a different search term",
  NO_DATA_AVAILABLE: "No data available",
  CLEAR_SYMBOL: "✕",
} as const;

// Colors
export const COLORS = {
  BUTTONS: {
    CLEAR: "#f44336",
    INSERT: "#4caf50",
    CRYPTO: "#2196f3",
    FIAT: "#ff9800",
    PURCHASABLE: "#9c27b0",
    DISABLED: "#cccccc",
    DEFAULT: "#007aff",
  },
  BACKGROUND: {
    PRIMARY: "#f5f5f5",
    WHITE: "#fff",
    MODAL: "#f5f5f5",
  },
  TEXT: {
    PRIMARY: "#333",
    SECONDARY: "#666",
    WHITE: "white",
    LINK: "#007aff",
  },
  CURRENCY_TYPE: {
    CRYPTO_BG: "#e3f2fd",
    CRYPTO_TEXT: "#1976d2",
    FIAT_BG: "#fff3e0",
    FIAT_TEXT: "#f57c00",
  },
  STATUS: {
    SUCCESS: "#4caf50",
    ERROR: "#f44336",
    WARNING: "#ff9800",
  },
} as const;

// API Endpoints
export const API_ENDPOINTS = {
  HEALTH: "/health",
  CURRENCIES: {
    CRYPTO: "/currencies/crypto",
    FIAT: "/currencies/fiat",
    ALL: "/currencies/all",
    SEARCH: (term: string) => `/currencies/search/${encodeURIComponent(term)}`,
    BY_ID: (id: string) => `/currencies/${encodeURIComponent(id)}`,
  },
} as const;
