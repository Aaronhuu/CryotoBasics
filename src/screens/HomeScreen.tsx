import React, { useState, useEffect } from "react";
import { Alert, Modal } from "react-native";
import { JSX } from "react";
import { CurrencyInfo } from "../types/CurrencyInfo";
import CurrencyListScreen from "../screens/CurrencyListScreen";
import LocalDatabase from "../services/LocalDatabase";
import ApiService from "../services/ApiService";
import {
  UI_TEXT,
  BUTTON_LABELS,
  MODAL_TITLES,
  ALERT_MESSAGES,
  SERVER_STATUS,
  COLORS,
} from "../constants";
import {
  ButtonContainer,
  ButtonText,
  CloseButton,
  CloseButtonText,
  Container,
  ModalContainer,
  ModalHeader,
  StyledButton,
  Subtitle,
  Title,
} from "../styles/styles";

export default function HomeScreen(): JSX.Element {
  const [databaseCount, setDatabaseCount] = useState(0);
  const [modalVisible, setModalVisible] = useState(false);
  const [currentList, setCurrentList] = useState<CurrencyInfo[]>([]);
  const [modalTitle, setModalTitle] = useState("");
  const [isLoading, setIsLoading] = useState(false);
  const [serverStatus, setServerStatus] = useState<
    (typeof SERVER_STATUS)[keyof typeof SERVER_STATUS]
  >(SERVER_STATUS.CHECKING);

  const database = LocalDatabase.getInstance();
  const apiService = ApiService.getInstance();

  useEffect(() => {
    updateDatabaseCount();
    checkServerStatus();
  }, []);

  const checkServerStatus = async () => {
    setServerStatus(SERVER_STATUS.CHECKING);
    try {
      const isRunning = await apiService.isServerRunning();
      setServerStatus(isRunning ? SERVER_STATUS.ONLINE : SERVER_STATUS.OFFLINE);
    } catch (error) {
      setServerStatus(SERVER_STATUS.OFFLINE);
    }
  };

  const updateDatabaseCount = () => {
    setDatabaseCount(database.getDataCount());
  };

  // Button 1: Clear data from local database
  const clearDatabase = async () => {
    setIsLoading(true);
    try {
      await database.clearAll();
      updateDatabaseCount();
      Alert.alert(
        ALERT_MESSAGES.DATABASE_CLEARED.title,
        ALERT_MESSAGES.DATABASE_CLEARED.message
      );
    } catch (error) {
      Alert.alert(
        ALERT_MESSAGES.ERRORS.CLEAR_DATABASE.title,
        ALERT_MESSAGES.ERRORS.CLEAR_DATABASE.message
      );
    } finally {
      setIsLoading(false);
    }
  };

  // Button 2: Insert data into local database
  const insertData = async () => {
    if (serverStatus === SERVER_STATUS.OFFLINE) {
      Alert.alert(
        ALERT_MESSAGES.SERVER_OFFLINE.title,
        ALERT_MESSAGES.SERVER_OFFLINE.message
      );
      return;
    }

    setIsLoading(true);
    try {
      const size = await database.insertAll();
      updateDatabaseCount();
      Alert.alert(
        ALERT_MESSAGES.DATA_INSERTED.title,
        ALERT_MESSAGES.DATA_INSERTED.message(size)
      );
    } catch (error) {
      Alert.alert(
        ALERT_MESSAGES.ERRORS.INSERT_DATA.title,
        ALERT_MESSAGES.ERRORS.INSERT_DATA.message
      );
      console.error("Insert data error:", error);
    } finally {
      setIsLoading(false);
    }
  };

  // Button 3: Show Currency List A - Crypto
  const showCryptoList = async () => {
    if (databaseCount === 0) {
      Alert.alert(ALERT_MESSAGES.NO_DATA.title, ALERT_MESSAGES.NO_DATA.message);
      return;
    }
    setIsLoading(true);
    try {
      const cryptoList = await database.getCryptoList();
      setCurrentList(cryptoList);
      setModalTitle(MODAL_TITLES.CRYPTO_LIST);
      setModalVisible(true);
    } catch (error) {
      Alert.alert(
        ALERT_MESSAGES.ERRORS.LOAD_CRYPTO.title,
        ALERT_MESSAGES.ERRORS.LOAD_CRYPTO.message
      );
    } finally {
      setIsLoading(false);
    }
  };

  // Button 4: Show Currency List B - Fiat
  const showFiatList = async () => {
    if (databaseCount === 0) {
      Alert.alert(ALERT_MESSAGES.NO_DATA.title, ALERT_MESSAGES.NO_DATA.message);
      return;
    }
    setIsLoading(true);
    try {
      const fiatList = await database.getFiatList();
      setCurrentList(fiatList);
      setModalTitle(MODAL_TITLES.FIAT_LIST);
      setModalVisible(true);
    } catch (error) {
      Alert.alert(
        ALERT_MESSAGES.ERRORS.LOAD_FIAT.title,
        ALERT_MESSAGES.ERRORS.LOAD_FIAT.message
      );
    } finally {
      setIsLoading(false);
    }
  };

  // Button 5: Show all purchasable currencies
  const showPurchasableCurrencies = async () => {
    if (databaseCount === 0) {
      Alert.alert(ALERT_MESSAGES.NO_DATA.title, ALERT_MESSAGES.NO_DATA.message);
      return;
    }
    setIsLoading(true);
    try {
      const purchasableList = await database.getAllList();
      setCurrentList(purchasableList);
      setModalTitle(MODAL_TITLES.PURCHASABLE_LIST);
      setModalVisible(true);
    } catch (error) {
      Alert.alert(
        ALERT_MESSAGES.ERRORS.LOAD_PURCHASABLE.title,
        ALERT_MESSAGES.ERRORS.LOAD_PURCHASABLE.message
      );
    } finally {
      setIsLoading(false);
    }
  };

  return (
    <Container>
      <Title>{UI_TEXT.APP_TITLE}</Title>
      <Subtitle>
        {UI_TEXT.DATABASE_STATUS(databaseCount, serverStatus)}
      </Subtitle>

      <ButtonContainer>
        <StyledButton
          testID="clear-database-button"
          onPress={clearDatabase}
          $color={COLORS.BUTTONS.CLEAR}
          $disabled={isLoading}
        >
          <ButtonText>
            {isLoading ? UI_TEXT.LOADING : BUTTON_LABELS.CLEAR_DATABASE}
          </ButtonText>
        </StyledButton>

        <StyledButton
          testID="insert-data-button"
          onPress={insertData}
          $color={COLORS.BUTTONS.INSERT}
          $disabled={isLoading}
        >
          <ButtonText>
            {isLoading ? UI_TEXT.LOADING : BUTTON_LABELS.INSERT_DATA}
          </ButtonText>
        </StyledButton>

        <StyledButton
          testID="show-crypto-list-button"
          onPress={showCryptoList}
          $color={COLORS.BUTTONS.CRYPTO}
          $disabled={isLoading}
        >
          <ButtonText>
            {isLoading ? UI_TEXT.LOADING : BUTTON_LABELS.SHOW_CRYPTO_LIST}
          </ButtonText>
        </StyledButton>

        <StyledButton
          testID="show-fiat-list-button"
          onPress={showFiatList}
          $color={COLORS.BUTTONS.FIAT}
          $disabled={isLoading}
        >
          <ButtonText>
            {isLoading ? UI_TEXT.LOADING : BUTTON_LABELS.SHOW_FIAT_LIST}
          </ButtonText>
        </StyledButton>

        <StyledButton
          testID="show-purchasable-list-button"
          onPress={showPurchasableCurrencies}
          $color={COLORS.BUTTONS.PURCHASABLE}
          $disabled={isLoading}
        >
          <ButtonText>
            {isLoading ? UI_TEXT.LOADING : BUTTON_LABELS.SHOW_PURCHASABLE}
          </ButtonText>
        </StyledButton>
      </ButtonContainer>

      <Modal
        animationType="slide"
        transparent={false}
        visible={modalVisible}
        onRequestClose={() => setModalVisible(false)}
      >
        <ModalContainer>
          <ModalHeader>
            <CloseButton onPress={() => setModalVisible(false)}>
              <CloseButtonText>{UI_TEXT.BACK}</CloseButtonText>
            </CloseButton>
          </ModalHeader>
          <CurrencyListScreen currencyList={currentList} title={modalTitle} />
        </ModalContainer>
      </Modal>
    </Container>
  );
}
