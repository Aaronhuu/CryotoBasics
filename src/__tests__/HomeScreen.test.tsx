import React from "react";
import { act, render, fireEvent } from "@testing-library/react-native";
import HomeScreen from "../screens/HomeScreen";
import { UI_TEXT, BUTTON_LABELS } from "../constants";
import { Alert } from "react-native";

jest.mock("@react-native-async-storage/async-storage", () => ({
  __esModule: true,
  default: {
    getItem: jest.fn(() => Promise.resolve(null)),
    setItem: jest.fn(() => Promise.resolve()),
    removeItem: jest.fn(() => Promise.resolve()),
    clear: jest.fn(() => Promise.resolve()),
  },
}));

// Mock the services
const mockClearAll = jest.fn().mockResolvedValue(undefined);
const mockGetDataCount = jest.fn().mockReturnValue(0);
const mockInsertAll = jest.fn().mockResolvedValue(10);
const mockGetCryptoList = jest.fn().mockResolvedValue([]);
const mockGetFiatList = jest.fn().mockResolvedValue([]);
const mockGetAllList = jest.fn().mockResolvedValue([]);

jest.mock("../services/LocalDatabase", () => ({
  __esModule: true,
  default: {
    getInstance: jest.fn(() => ({
      getDataCount: mockGetDataCount,
      clearAll: mockClearAll,
      insertAll: mockInsertAll,
      getCryptoList: mockGetCryptoList,
      getFiatList: mockGetFiatList,
      getAllList: mockGetAllList,
    })),
  },
}));

jest.mock("../services/ApiService", () => ({
  __esModule: true,
  default: {
    getInstance: jest.fn(() => ({
      isServerRunning: jest.fn().mockResolvedValue(true),
    })),
  },
}));

describe("HomeScreen UI", () => {
  beforeEach(() => {
    // Reset mocks before each test
    jest.clearAllMocks();
    jest.spyOn(Alert, "alert").mockImplementation(() => {});
  });

  it("should render without crashing", () => {
    // This test verifies that the HomeScreen component can be rendered
    // without throwing any errors, which validates the basic UI structure
    const { toJSON, getByText } = render(<HomeScreen />);
    expect(toJSON()).toMatchSnapshot();

    expect(getByText(UI_TEXT.APP_TITLE)).toBeTruthy();
    expect(getByText(BUTTON_LABELS.CLEAR_DATABASE)).toBeTruthy();
    expect(getByText(BUTTON_LABELS.INSERT_DATA)).toBeTruthy();
    expect(getByText(BUTTON_LABELS.SHOW_CRYPTO_LIST)).toBeTruthy();
    expect(getByText(BUTTON_LABELS.SHOW_FIAT_LIST)).toBeTruthy();
    expect(getByText(BUTTON_LABELS.SHOW_PURCHASABLE)).toBeTruthy();
  });

  it("should handle clear button press", async () => {
    const { getByTestId } = render(<HomeScreen />);
    const clearButton = getByTestId("clear-database-button");

    // Press the clear button
    await act(async () => {
      fireEvent.press(clearButton);
    });

    // Verify that the clearAll method was called
    expect(mockClearAll).toHaveBeenCalledTimes(1);

    // Verify that getDataCount was called to update the count
    expect(mockGetDataCount).toHaveBeenCalled();
    expect(Alert.alert).toHaveBeenCalledWith(
      "Database Cleared",
      "All data has been removed from the local database."
    );
  });

  it("should handle insert data button press", async () => {
    const { getByTestId } = render(<HomeScreen />);
    const insertButton = getByTestId("insert-data-button");

    // Press the insert data button
    await act(async () => {
      fireEvent.press(insertButton);
    });

    // Verify that the insertAll method was called
    expect(mockInsertAll).toHaveBeenCalledTimes(1);
    expect(mockGetDataCount).toHaveBeenCalled();
    expect(Alert.alert).toHaveBeenCalledWith(
      "Data Inserted",
      "10 currency records have been inserted into the local database."
    );
  });

  it("should handle show crypto list button press", async () => {
    // Set database count to 1 so the button works
    mockGetDataCount.mockReturnValue(1);

    const { getByTestId } = render(<HomeScreen />);
    const cryptoButton = getByTestId("show-crypto-list-button");

    // Press the crypto list button
    await act(async () => {
      fireEvent.press(cryptoButton);
    });

    // Verify that getCryptoList was called
    expect(mockGetCryptoList).toHaveBeenCalledTimes(1);
  });

  it("should handle show fiat list button press", async () => {
    // Set database count to 1 so the button works
    mockGetDataCount.mockReturnValue(1);

    const { getByTestId } = render(<HomeScreen />);
    const fiatButton = getByTestId("show-fiat-list-button");

    // Press the fiat list button
    await act(async () => {
      fireEvent.press(fiatButton);
    });

    // Verify that getFiatList was called
    expect(mockGetFiatList).toHaveBeenCalledTimes(1);
  });

  it("should handle show purchasable list button press", async () => {
    // Set database count to 1 so the button works
    mockGetDataCount.mockReturnValue(1);

    const { getByTestId } = render(<HomeScreen />);
    const purchasableButton = getByTestId("show-purchasable-list-button");

    // Press the purchasable list button
    await act(async () => {
      fireEvent.press(purchasableButton);
    });

    // Verify that getAllList was called
    expect(mockGetAllList).toHaveBeenCalledTimes(1);
  });

  it("should show alert when trying to show lists with empty database", async () => {
    // Ensure database count is 0
    mockGetDataCount.mockReturnValue(0);

    const { getByTestId } = render(<HomeScreen />);
    const cryptoButton = getByTestId("show-crypto-list-button");

    // Press the crypto list button
    await act(async () => {
      fireEvent.press(cryptoButton);
    });

    // Should show no data alert instead of calling getCryptoList
    expect(mockGetCryptoList).not.toHaveBeenCalled();
    expect(Alert.alert).toHaveBeenCalledWith(
      "No Data",
      "Please insert data first using the 'Insert Data' button."
    );
  });

  it("should handle database clear error", async () => {
    // Mock clearAll to reject
    mockClearAll.mockRejectedValueOnce(new Error("Clear failed"));

    const { getByTestId } = render(<HomeScreen />);
    const clearButton = getByTestId("clear-database-button");

    // Press the clear button
    await act(async () => {
      fireEvent.press(clearButton);
    });

    // Should show error alert
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Failed to clear database."
    );
  });

  it("should handle insert data error", async () => {
    // Mock insertAll to reject
    mockInsertAll.mockRejectedValueOnce(new Error("Insert failed"));

    const { getByTestId } = render(<HomeScreen />);
    const insertButton = getByTestId("insert-data-button");

    // Press the insert button
    await act(async () => {
      fireEvent.press(insertButton);
    });

    // Should show error alert
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Failed to insert data. Make sure the server is running."
    );
  });

  it("should handle crypto list loading error", async () => {
    // Set database count to 1 and mock getCryptoList to reject
    mockGetDataCount.mockReturnValue(1);
    mockGetCryptoList.mockRejectedValueOnce(new Error("Load failed"));

    const { getByTestId } = render(<HomeScreen />);
    const cryptoButton = getByTestId("show-crypto-list-button");

    // Press the crypto list button
    await act(async () => {
      fireEvent.press(cryptoButton);
    });

    // Should show error alert
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Failed to load crypto currencies."
    );
  });

  it("should handle fiat list loading error", async () => {
    // Set database count to 1 and mock getFiatList to reject
    mockGetDataCount.mockReturnValue(1);
    mockGetFiatList.mockRejectedValueOnce(new Error("Load failed"));

    const { getByTestId } = render(<HomeScreen />);
    const fiatButton = getByTestId("show-fiat-list-button");

    // Press the fiat list button
    await act(async () => {
      fireEvent.press(fiatButton);
    });

    // Should show error alert
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Failed to load fiat currencies."
    );
  });

  it("should handle purchasable list loading error", async () => {
    // Set database count to 1 and mock getAllList to reject
    mockGetDataCount.mockReturnValue(1);
    mockGetAllList.mockRejectedValueOnce(new Error("Load failed"));

    const { getByTestId } = render(<HomeScreen />);
    const purchasableButton = getByTestId("show-purchasable-list-button");

    // Press the purchasable list button
    await act(async () => {
      fireEvent.press(purchasableButton);
    });

    // Should show error alert
    expect(Alert.alert).toHaveBeenCalledWith(
      "Error",
      "Failed to load purchasable currencies."
    );
  });

  it("should display correct database status text", () => {
    mockGetDataCount.mockReturnValue(5);

    const { getByText } = render(<HomeScreen />);

    // Check that database status is displayed correctly
    expect(getByText(/Database: 5 records \| Server: checking/)).toBeTruthy();
  });

  it("should disable buttons when loading", async () => {
    const { getByTestId, getAllByText } = render(<HomeScreen />);
    const clearButton = getByTestId("clear-database-button");

    // Start an operation that sets loading state
    act(() => {
      fireEvent.press(clearButton);
    });

    // Check that the button text changes to "Loading..."
    expect(getAllByText("Loading...")).toBeTruthy();
  });
});
