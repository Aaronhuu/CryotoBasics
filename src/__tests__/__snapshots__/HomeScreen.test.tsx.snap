// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen UI should render without crashing 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#f5f5f5",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "justifyContent": "center",
      "paddingBottom": 20,
      "paddingLeft": 20,
      "paddingRight": 20,
      "paddingTop": 20,
    }
  }
>
  <Text
    style={
      {
        "color": "#333",
        "fontSize": 28,
        "fontWeight": "bold",
        "marginBottom": 10,
        "textAlign": "center",
      }
    }
  >
    Crypto Basics Demo
  </Text>
  <Text
    style={
      {
        "color": "#666",
        "fontSize": 16,
        "marginBottom": 30,
        "textAlign": "center",
      }
    }
  >
    Database: 0 records | Server: checking
  </Text>
  <View
    style={
      {
        "alignItems": "center",
        "gap": 12,
        "width": "100%",
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#f44336",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "elevation": 5,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 250,
        }
      }
      testID="clear-database-button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Clear Database
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#4caf50",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "elevation": 5,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 250,
        }
      }
      testID="insert-data-button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Insert Data
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#2196f3",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "elevation": 5,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 250,
        }
      }
      testID="show-crypto-list-button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Show Crypto List
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ff9800",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "elevation": 5,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 250,
        }
      }
      testID="show-fiat-list-button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Show Fiat List
      </Text>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#9c27b0",
          "borderBottomLeftRadius": 8,
          "borderBottomRightRadius": 8,
          "borderTopLeftRadius": 8,
          "borderTopRightRadius": 8,
          "elevation": 5,
          "opacity": 1,
          "paddingBottom": 15,
          "paddingLeft": 30,
          "paddingRight": 30,
          "paddingTop": 15,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 3.84,
          "width": 250,
        }
      }
      testID="show-purchasable-list-button"
    >
      <Text
        style={
          {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
          }
        }
      >
        Show Purchasable
      </Text>
    </View>
  </View>
</View>
`;
